### PR TITLE
Add WASM tail calls to Fx 121 release notes

### PR DESCRIPTION
**Pull request by pepelsbey**
### Description

Add WASM tail calls to Fx 121 release notes. 

### Related issues and pull requests

Part of the Firefox 121 release task https://github.com/mdn/content/issues/30333

### CODE REVIEW

File: files/en-us/mozilla/firefox/releases/121/index.md [hunk 1]
```diff
@@ -74,6 +74,8 @@ This article provides information about the changes in Firefox 121 that affect d
 
 ### WebAssembly
 
+- WebAssembly now supports tail call optimization via new `return_call` and `return_call_indirect` alternatives to the [`call`](/en-US/docs/WebAssembly/Reference/Control_flow/call) instruction. This improves performance, reduces stack space usage, and enhances compatibility with programming languages. ([Firefox bug 1858855](https://bugzil.la/1846789)).
```
Comment (zfox23) [inline files/en-us/mozilla/firefox/releases/121/index.md:77]: In just a couple of extra words, is there a way for you to further explain what you mean by "enhances compatibility with programming languages"? I don't quite understand that point.

**Review by zfox23 — CHANGES_REQUESTED** [review]
Thanks! Curious if you could clarify one thing... If there's no language you can think of for this, that's OK too.

File: files/en-us/mozilla/firefox/releases/121/index.md (hunk 1 above)
Comment (pepelsbey) [reply files/en-us/mozilla/firefox/releases/121/index.md]: > and enhances compatibility with programming languages **that use tail calls**

I added a small note that hopefully makes it a bit clearer. I didn’t want to go too deep into details explaining how the whole WASM thing works.

**Review by zfox23 — APPROVED** [review]

**Comment by zfox23** [comment]
ty :)

**Pull request merged**